src/main/java/org/extism/sdk/chicory/Plugin.java

```diff
@@ -83,7 +83,11 @@ String getError() {
     public byte[] call(String funcName, byte[] input) {
         var func = mainInstance.export(funcName);
         setInput(input);
-        var result = func.apply()[0];
+        var results = func.apply();
+        if (results == null) {
+            throw new ExtismFunctionException(funcName, "The function expects an i32 return code. 0 is success and 1 is a failure.");
+        }
+        var result = results[0];
```

Comment:
thanks, great point. I think the test case can be simpler though. Something like


```wat
(module
    (func $dummy (export "dummy"))
)
```